src/vcModals.cpp

```diff
@@ -1815,6 +1815,30 @@ void vcModals_DrawImportShapeFile(vcState *pProgramState)
   }
 }
 
+void vcModals_DrawFlythroughExport(vcState *pProgramState)
+{
+  if (pProgramState->openModals & (1 << vcMT_FlythroughExport))
+  {
+    ImGui::OpenPopup("###modalFlythroughExport");
+    ImGui::SetNextWindowSize(ImVec2(210, 70), ImGuiCond_FirstUseEver);
+  }
+
+  if (ImGui::BeginPopupModal("###modalFlythroughExport", nullptr, ImGuiWindowFlags_NoTitleBar))
+  {
+    if (pProgramState->closeModals & (1 << vcMT_FlythroughExport))
+      ImGui::CloseCurrentPopup();
+    else
+      pProgramState->modalOpen = true;
+
+    ImGui::TextUnformatted(vcString::Get("flythroughExporting"));
+
+    if (ImGui::Button(vcString::Get("flythroughExportCancel"), ImVec2(200, 30)))
+      pProgramState->flythroughExportCancel = true;
```

Comment:
Alternatively, you could add a function to the `vcFlythrough` scene item, and call that instead? The selected item is always going to be a flythrough since selecting another scene item will cancel the flythrough.
```c++
vcFlythrough *pFlythrough = (vcFlythrough*)pProgramState->sceneExplorer.clickedItem.pItem->pUserData;
```

Reply:
i like avoiding clogging up state, but there only ever needs to be 1 instance of this variable. doesn't make sense for each flythrough to have their own cancel variable.

Reply:
As discussed on teams, I meant a cancel function rather than a new variable, this would remove the additional state created here in favour of just performing the cancel action.